src/Microsoft.IdentityModel.Tokens/AuthenticationTicket/MicrosoftClaimsIdentity.cs

```diff
@@ -0,0 +1,50 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Security.Claims;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Extension of ClaimsIdentity to represent the data like the Token, the raw token
+    /// </summary>
+    public class MicrosoftClaimsIdentity : ClaimsIdentity 
+    {
+        // TODO: decide the contructors to have.
```

Comment:
@jennyf19 
Perhaps we can incorporate these changes to improve the accessing of claims, currently ClaimsIdentity has a O(n) linear profile.
By overloading virtuals such as: 
public virtual IEnumerable<Claim> Claims
protected virtual byte[]? CustomSerializationData
public virtual void AddClaim(Claim claim)
public virtual void AddClaims(IEnumerable<Claim?> claims)
public virtual bool TryRemoveClaim(Claim? claim)
etc..
see: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs